packages/test/test-end-to-end-tests/src/test/directoryEndToEndTests.spec.ts

```diff
@@ -1343,21 +1347,35 @@ describeCompat(
 			expectSubdirsOrder(sharedDirectory3, dirsInOrder, path);
 		}
 
-		// This test falsely assumes ops will be acked in the order they're submtited.
-		// AB#6515 tracks re-enabling.
-		it.skip("Eventual consistency in ordering with subdirectories creation/deletion", async () => {
+		async function pauseAllContainersInbound() {
+			await container1.deltaManager.inbound.pause();
+			await container2.deltaManager.inbound.pause();
+			await container3.deltaManager.inbound.pause();
+		}
+
+		function resumeAllContainersInbound() {
+			container1.deltaManager.inbound.resume();
+			container2.deltaManager.inbound.resume();
+			container3.deltaManager.inbound.resume();
+		}
+
+		it("Eventual consistency in ordering with subdirectories creation/deletion", async () => {
+			// Pause to not allow ops to be processed while we maintained them in order.
+			await pauseAllContainersInbound();
+
 			sharedDirectory1.createSubDirectory("dir2");
 			sharedDirectory2.createSubDirectory("dir1");
 			sharedDirectory2.createSubDirectory("dir2");
 			sharedDirectory3.createSubDirectory("dir3");
 			sharedDirectory3.createSubDirectory("dir2");
 
-			await provider.opProcessingController.processIncoming();
 			expectSubdirsOrder(sharedDirectory1, ["dir2"]);
 			expectSubdirsOrder(sharedDirectory2, ["dir1", "dir2"]);
 			expectSubdirsOrder(sharedDirectory3, ["dir3", "dir2"]);
 
-			await provider.opProcessingController.processOutgoing();
+			resumeAllContainersInbound();
```

Comment:
AFAIK there's still nothing to ensure that "dir2" is created (i.e. sequenced) before "dir1". You need such mechanisms for this test to be deterministic, e.g. strategically unpausing and waiting for ack of containers in some order.